src/Database/QueryBuilder.php

```diff
@@ -181,4 +208,72 @@ public function count($columns = '*')
 
         return $result;
     }
+
+    /**
+     * Update a record in the database.
+     *
+     * @param  array $values
+     * @return int
+     */
+    public function update(array $values)
+    {
+        $this->clearMemoryCache($this->from);
+
+        return parent::update($values);
+    }
+
+    /**
+     * Delete a record from the database.
+     *
+     * @param  mixed $id
+     * @return int
+     */
+    public function delete($id = null)
+    {
+        $this->clearMemoryCache($this->from);
+
+        return parent::delete($id);
+    }
+
+    /**
+     * Insert a new record into the database.
+     *
+     * @param  array  $values
+     * @return bool
+     */
+    public function insert(array $values)
+    {
+        $this->clearMemoryCache($this->from);
+
+        return parent::insert($values);
+    }
+
+    /**
+     * Run a truncate statement on the table.
+     *
+     * @return void
+     */
+    public function truncate()
+    {
+        $this->clearMemoryCache($this->from);
+
+        parent::truncate();
+    }
+
+    /**
+     * Clear the memory cache.
+     *
+     * @param  string|null  $table
+     * @return \Illuminate\Database\Query\Builder|static
+     */
+    public function clearMemoryCache($table = null)
+    {
+        $cache = MemoryCache::instance();
+
+        if ($cache->enabled()) {
+            $table ? $cache->forget($table) : $cache->flush();
```

Comment:
Do you mind explaining the rational behind this? I'm not sure that clearing the entire cache from a call to the query builder is a good idea, I preferred the other method of defaulting to the current table. If you want to have a method to clear the entire cache, then perhaps it should be separate from this one or evaluated as a parameter passed here.